fix(curriculum): Merge Sort algorithm step 10 requires spaces to pass

subhadipjana866 opened the pull request:
This pull request fixes the issue number #55977 by adding /s* in the regular expressions.  The issue is fixed.

Fix #55977

Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #55977

<!-- Feel free to add any additional description of changes below this line -->

subhadipjana866 committed:
fix(curriculum): Merge Sort algorithm step 10 requires spaces to pass

This pull request fixes the issue number #55977 by adding /s* in the regular expressions. 
The issue is fixed.

Fix #55977

lasjorg commented on curriculum/challenges/english/07-scientific-computing-with-python/learn-data-structures-by-building-the-merge-sort-algorithm/656680b0fc79f2c38a34d90e.md:
```diff
@@ -24,7 +24,7 @@ You should have a variable named `left_array_index` inside your `merge_sort` fun
     const { function_body } = merge_sort;
 
-    assert.match(function_body, /left_array_index\s*(?!=)/);
+    assert.match(function_body, /left_array_index\s*(?!=)\s*/);
```
```suggestion
    assert.match(function_body, /left_array_index/);
```

lasjorg commented on curriculum/challenges/english/07-scientific-computing-with-python/learn-data-structures-by-building-the-merge-sort-algorithm/656680b0fc79f2c38a34d90e.md:
```diff
@@ -52,7 +52,7 @@ You should have a variable named `right_array_index` inside your `merge_sort` fu
     const { function_body } = merge_sort;
 
-    assert.match(function_body, /right_array_index\s*(?!=)/);
+    assert.match(function_body, /right_array_index\s*(?!=)\s*/);
```
```suggestion
    assert.match(function_body, /right_array_index/);
```

lasjorg commented on curriculum/challenges/english/07-scientific-computing-with-python/learn-data-structures-by-building-the-merge-sort-algorithm/656680b0fc79f2c38a34d90e.md:
```diff
@@ -81,7 +81,7 @@ You should have a variable named `sorted_index` inside your `merge_sort` functio
     const { function_body } = merge_sort;
 
-    assert.match(function_body, /sorted_index\s*(?!=)/);
+    assert.match(function_body, /sorted_index\s*(?!=)\s*/);
```
```suggestion
    assert.match(function_body, /sorted_index/);
```

lasjorg reviewed (changes requested):
I know you didn't write it, but I don't understand the negative look ahead. It makes the optional space required (as far as I can tell).

Considering identifiers without an assignment is valid syntax, but an assignment without a value isn't, wouldn't it make more sense to just have the first regex for each identifier only look for the identifier?

subhadipjana866 committed:
Update curriculum/challenges/english/07-scientific-computing-with-python/learn-data-structures-by-building-the-merge-sort-algorithm/656680b0fc79f2c38a34d90e.md

My solution was to check for zero or more spaces before and after the '=' sign. Yeah the suggested solution does work. Thanks.

Co-authored-by: Lasse Jørgensen <28780271+lasjorg@users.noreply.github.com>

subhadipjana866 committed:
Update curriculum/challenges/english/07-scientific-computing-with-python/learn-data-structures-by-building-the-merge-sort-algorithm/656680b0fc79f2c38a34d90e.md

same for this

Co-authored-by: Lasse Jørgensen <28780271+lasjorg@users.noreply.github.com>

subhadipjana866 committed:
Update curriculum/challenges/english/07-scientific-computing-with-python/learn-data-structures-by-building-the-merge-sort-algorithm/656680b0fc79f2c38a34d90e.md

Co-authored-by: Lasse Jørgensen <28780271+lasjorg@users.noreply.github.com>

lasjorg reviewed (approved):
LGTM

moT01 reviewed (approved):
LGTM @subhadipjana866 🎉 Congratulations on your first contribution to the freeCodeCamp repo 🎉
